Code/Tools/AssetProcessor/native/unittests/RCcontrollerUnitTests.cpp

```diff
@@ -214,315 +158,455 @@ void RCcontrollerUnitTests::RunRCControllerTests()
+    m_rcController->m_RCQueueSortModel.AttachToModel(nullptr);
+    m_rcController->m_RCQueueSortModel.AttachToModel(&m_rcController->m_RCJobListModel);
+    m_rcController->m_RCQueueSortModel.m_currentJobRunKeyToJobEntries.clear();
+    m_rcController->m_RCQueueSortModel.m_currentlyConnectedPlatforms.clear();
+}
+
+void RCcontrollerUnitTests::ConnectCompileGroupSignalsAndSlots(bool& gotCreated, bool& gotCompleted, NetworkRequestID& gotGroupID, AssetStatus& gotStatus)
+{
+    QObject::connect(m_rcController.get(), &RCController::CompileGroupCreated, this, [&](NetworkRequestID groupID, AssetStatus status)
         {
             gotCreated = true;
             gotGroupID = groupID;
             gotStatus = status;
         });
 
-    QObject::connect(&m_rcController, &RCController::CompileGroupFinished, this, [&](NetworkRequestID groupID, AssetStatus status)
+    QObject::connect(m_rcController.get(), &RCController::CompileGroupFinished, this, [&](NetworkRequestID groupID, AssetStatus status)
         {
             gotCompleted = true;
             gotGroupID = groupID;
             gotStatus = status;
         });
+}
+
+void RCcontrollerUnitTests::ConnectJobSignalsAndSlots(bool& allJobsCompleted, JobEntry& completedJob)
+{
+    QObject::connect(m_rcController.get(), &RCController::FileCompiled, this, [&](JobEntry entry, AssetBuilderSDK::ProcessJobResponse response)
+        {
+            completedJob = entry;
+        });
+
+    QObject::connect(m_rcController.get(), &RCController::FileCancelled, this, [&](JobEntry entry)
+        {
+            completedJob = entry;
+        });
+
+    QObject::connect(m_rcController.get(), &RCController::FileFailed, this, [&](JobEntry entry)
+        {
+            completedJob = entry;
+        });
+    QObject::connect(m_rcController.get(), &RCController::ActiveJobsCountChanged, this, [&](unsigned int /*count*/)
+        {
+            m_rcController->OnAddedToCatalog(completedJob);
+            completedJob = {};
+        });
+
+    QObject::connect(m_rcController.get(), &RCController::BecameIdle, this, [&]()
+        {
+            allJobsCompleted = true;
+        }
+    );
+}
+
+void RCcontrollerUnitTests::SetUp()
+{
+    UnitTest::AssetProcessorUnitTestBase::SetUp();
+    m_rcController = AZStd::make_unique<AssetProcessor::RCController>(1, 4);
+
+    using namespace AssetProcessor;
+    m_rcJobListModel = m_rcController->GetQueueModel();
+    m_rcQueueSortModel = &m_rcController->m_RCQueueSortModel;
+}
 
+void RCcontrollerUnitTests::TearDown()
+{
+    m_rcJobListModel = nullptr;
+    m_rcQueueSortModel = nullptr;
+
+    m_rcController.reset();
+
+    UnitTest::AssetProcessorUnitTestBase::TearDown();
+}
+
+TEST_F(RCcontrollerUnitTests, TestRCJobListModel_AddJobEntries_Succeeds)
+{
+    PrepareRCJobListModelTest();
+
+    int returnedCount = m_rcJobListModel->rowCount(QModelIndex());
+    int expectedCount = 5; // finished ones should be removed, so it shouldn't show up
```

Comment:
nit: Be clear of the what, maybe "Finished jobs should be removed, so they shouldn't show up".

Provide guidance as to why 5 for future maintainers.